### DISCUSSION TITLE
Support s3ForcePathStyle option for S3 File Storage

### DISCUSSION THREAD

**Discussion by mischal** [directus/directus#4632] [Feature Requests]
If we can set the s3ForcePathStyle option in the File Storage configuration other S3-Compliant Object Storage Services can be supported (for example [Minio](https://github.com/minio/minio))

Suggestion:
New boolean configuration Key for the S3 File Storage option: `STORAGE_<LOCATION>_FORCE_PATH_STYLE`
and pass as `s3ForcePathStyle` to the S3 driver (https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#s3ForcePathStyle-property)

Can I write a PR for this small improvement?

**Comment by rijkvanzanten** [comment]
This already works 🙂 All keys are passed on to the AWS SDK config

**Reply by mischal** [reply]
Oh, I didnt know that. Thats very nice :-) 👍  How can I use the key in the .env file? is this `STORAGE_<LOCATION>_s3ForcePathStyle` or are there any translations regarding the case styles?

**Reply by rijkvanzanten** [reply]
It automatically converts the underscores to camelcase 🙂 So it'd be `STORAGE_<LOCATION>_S3_FORCE_PATH_STYLE`

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
